internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -388,27 +383,55 @@ func (q amsQuotaService) DeleteQuotaForBillingModel(subscriptionID string, kafka
 // return an error if an error happens calling the endpoint, otherwise returns whether it found the subscription(subscription, true, nil)
 // or not (subscription, false, nil)
 func (q amsQuotaService) GetSubscriptionByID(subscriptionID string) (*amsv1.Subscription, bool, *errors.ServiceError) {
-	resp, err := q.amsClient.GetSubscriptionByID(subscriptionID)
-	if resp != nil && resp.Status() == http.StatusNotFound {
-		return resp.Body(), false, nil
-	}
+	subscription, ok, err := q.amsClient.GetSubscriptionByID(subscriptionID)
 	if err != nil {
 		return nil, false, errors.NewWithCause(errors.ErrorGeneral, err, "error getting the subscriptions for id '%s'", subscriptionID)
 	}
 
-	return resp.Body(), true, nil
+	return subscription, ok, nil
 }
 
-// Checks if an organisation has a SKU entitlement and that entitlement is still active in AMS
+func (q amsQuotaService) amsBillingModelToAMSBillingModelCategory(amsBillingModel string, kafkaBillingModel config.KafkaBillingModel) (string, bool) {
```

Comment:
This method name can be confusing, can we add a comment describing this method explaining what is 'ams billing model' and 'ams billing model category'